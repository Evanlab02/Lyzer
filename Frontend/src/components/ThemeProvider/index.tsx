import { useEffect, useState } from "react";
import { ThemeContext } from "../../hooks/useTheme";

export const ThemeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
	const defaultDark = window.matchMedia("(prefers-color-scheme: dark)").matches;

	const [isDarkMode, setDarkMode] = useState(
		localStorage.getItem("isDarkMode") !== null
			? localStorage.getItem("isDarkMode") === "true"
			: defaultDark
	);

	const toggleTheme = () => {
		setDarkMode((prev) => !prev);
	};

	useEffect(() => {
		const theme = isDarkMode ? "dark" : "light";
		localStorage.setItem("isDarkMode", theme === "dark" ? "true" : "false");
		document.documentElement.setAttribute("data-theme", theme);
	}, [isDarkMode]);

	return (
		<ThemeContext.Provider value={{ isDarkMode, toggleTheme }}>
			{children}
		</ThemeContext.Provider>
	);
};