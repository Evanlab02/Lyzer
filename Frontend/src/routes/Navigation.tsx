import { Route, Routes } from "react-router-dom";
import Overview from "../pages/Overview";
import NotFound from "../pages/404";

export default function Navigation() {
	return (
		<Routes>
			<Route path="" element={<Overview/>} />
			<Route path="*" element={<NotFound/>} />
		</Routes>
	);
};
