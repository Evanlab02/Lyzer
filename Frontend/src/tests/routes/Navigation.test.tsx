import { expect, it } from "vitest";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import ThemeProvider from "../../components/ThemeProvider";
import Navigation from "../../routes/Navigation";


it("Navigation renders overview page correctly", async () => {
	const { findByTestId } = render(
		<MemoryRouter initialEntries={["/"]}>
			<ThemeProvider>
				<Navigation />
			</ThemeProvider>
		</MemoryRouter>
	);

	const overviewPage = await findByTestId("lyzer-overview-page");
	expect(overviewPage).toMatchSnapshot();
});

it("Navigation renders 404 page correctly", async () => {
	const { findByTestId } = render(
		<MemoryRouter initialEntries={["/non-existent-route"]}>
			<ThemeProvider>
				<Navigation />
			</ThemeProvider>
		</MemoryRouter>
	);

	const notFoundPage = await findByTestId("lyzer-not-found-page");
	expect(notFoundPage).toMatchSnapshot();
});
