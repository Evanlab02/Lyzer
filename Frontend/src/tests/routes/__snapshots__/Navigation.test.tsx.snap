// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation renders 404 page correctly 1`] = `
<div
  data-testid="lyzer-not-found-page"
>
  Page not found
</div>
`;

exports[`Navigation renders overview page correctly 1`] = `
<div
  class="page-content"
  data-testid="lyzer-overview-page"
>
  <div
    class="flex-row justify-between items-center"
  >
    <h2>
      Overview
    </h2>
    <svg
      class="lucide lucide-refresh-ccw cursor-pointer"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 0 0-9-9 9.75 9.75 0 0 0-6.74 2.74L3 8"
      />
      <path
        d="M3 3v5h5"
      />
      <path
        d="M3 12a9 9 0 0 0 9 9 9.75 9.75 0 0 0 6.74-2.74L21 16"
      />
      <path
        d="M16 16h5v5"
      />
    </svg>
  </div>
  <div
    class="flex-grid-container"
  >
    <div
      class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-6 flex-grid-col-lg-4 flex-grid-col-xl-3 flex-grid-col-xxl-3"
    >
      <div
        class="lyzer-card"
      >
        <span
          class="lyzer-card-heading"
        >
          Is it race weekend?
        </span>
        <h3
          class="lyzer-card-body"
        >
          No
          <p
            class="text-sm m-0"
          >
            3 days to go
          </p>
        </h3>
        <progress
          class="lyzer-progress lyzer-progress-incomplete"
          max="100"
          value="50"
        />
      </div>
    </div>
    <div
      class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-6 flex-grid-col-lg-4 flex-grid-col-xl-3 flex-grid-col-xxl-3"
    >
      <div
        class="lyzer-card"
      >
        <span
          class="lyzer-card-heading"
        >
          Race weekend progress
        </span>
        <h3
          class="lyzer-card-body"
        >
          Next session: Practice 1
          <p
            class="text-sm m-0"
          >
            Sat, Feb 22, 2025, 12:00 PM
          </p>
        </h3>
        <progress
          class="lyzer-progress lyzer-progress-incomplete"
          max="100"
          value="0"
        />
      </div>
    </div>
    <div
      class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-6 flex-grid-col-lg-4 flex-grid-col-xl-3 flex-grid-col-xxl-3"
    >
      <div
        class="lyzer-card"
      >
        <span
          class="lyzer-card-heading"
        >
          Driver leader
        </span>
      </div>
    </div>
    <div
      class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-6 flex-grid-col-lg-4 flex-grid-col-xl-3 flex-grid-col-xxl-3"
    >
      <div
        class="lyzer-card"
      >
        <span
          class="lyzer-card-heading"
        >
          Constructor leader
        </span>
      </div>
    </div>
    <div
      class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-6 flex-grid-col-lg-4 flex-grid-col-xl-3 flex-grid-col-xxl-3"
    >
      <div
        class="lyzer-card"
      >
        <h1>
          Overview Section
        </h1>
      </div>
    </div>
    <div
      class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-6 flex-grid-col-lg-4 flex-grid-col-xl-4 flex-grid-col-xxl-4"
    >
      <div
        class="lyzer-card"
      >
        <h1>
          Constructor Standings
        </h1>
      </div>
    </div>
    <div
      class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-6 flex-grid-col-lg-4 flex-grid-col-xl-5 flex-grid-col-xxl-5"
    >
      <div
        class="lyzer-card"
      >
        <h1>
          Driver Standings
        </h1>
      </div>
    </div>
  </div>
</div>
`;
